composer/utils/checkpoint.py

```diff
@@ -400,11 +400,24 @@ def read_data(self, plan: LoadPlan, planner: LoadPlanner):
 
         # We need no_grad because we overwrite tensor values with set_() when we do elastic loading and we don't want the set_ op recorded in the computation graph.
         with torch.no_grad():
-            # 1. Load just model first.
-            model_state_dict = {'state': {'model': state.state_dict()['model']}}
+            # 1. Load model and metadata first
+            model_state_dict = None
+            if load_weights_only:
+                model_state_dict = {'state': {'model': state.get_model_state_dict()}}
+            else:
+                cur_state_dict = state.state_dict()
```

Comment:
I wish there was a way to do it without materializing the optimizer state

Reply:
Is it slow? do I have to be concerned about it?